src/mongo_logger.ts

```diff
@@ -124,56 +176,247 @@ function resolveLogPath(
   }: {
     mongodbLogPath?: unknown;
   }
-): Writable {
+): MongoDBLogWritable {
   const isValidLogDestinationString = (destination: string) =>
     ['stdout', 'stderr'].includes(destination.toLowerCase());
   if (typeof mongodbLogPath === 'string' && isValidLogDestinationString(mongodbLogPath)) {
-    return mongodbLogPath.toLowerCase() === 'stderr' ? process.stderr : process.stdout;
+    return (mongodbLogPath.toLowerCase() === 'stderr'
+      ? process.stderr
+      : process.stdout) as unknown as MongoDBLogWritable;
   }
 
-  // TODO(NODE-4813): check for minimal interface instead of instanceof Writable
-  if (typeof mongodbLogPath === 'object' && mongodbLogPath instanceof Writable) {
-    return mongodbLogPath;
+  if (
+    typeof mongodbLogPath === 'object' &&
+    // eslint-disable-next-line no-restricted-syntax
+    mongodbLogPath !== null &&
+    Object.prototype.hasOwnProperty.call(mongodbLogPath, 'write')
+  ) {
+    return mongodbLogPath as MongoDBLogWritable;
   }
 
   if (typeof MONGODB_LOG_PATH === 'string' && isValidLogDestinationString(MONGODB_LOG_PATH)) {
-    return MONGODB_LOG_PATH.toLowerCase() === 'stderr' ? process.stderr : process.stdout;
+    return (MONGODB_LOG_PATH.toLowerCase() === 'stderr'
+      ? process.stderr
+      : process.stdout) as unknown as MongoDBLogWritable;
   }
 
-  return process.stderr;
+  return process.stderr as unknown as MongoDBLogWritable;
+}
+
+/** @internal */
+export interface Log extends Record<string, any> {
+  s: SeverityLevel;
+  t: Date;
+  c: MongoLoggableComponent;
+  message?: string;
+}
+
+/** @internal */
+export interface MongoDBLogWritable {
+  write(log: Log): unknown;
+}
```

Comment:
Is this interface too simplistic? Should we explicitly accept a union of this interface and an actual `Writable` stream?